keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/operators/multiple/ProcessModelVertex.java

```diff
@@ -0,0 +1,114 @@
+package io.improbable.keanu.vertices.dbl.nonprobabilistic.operators.multiple;
+
+import io.improbable.keanu.tensor.Tensor;
+import io.improbable.keanu.tensor.bool.BooleanTensor;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.tensor.intgr.IntegerTensor;
+import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.VertexLabel;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * A vertex whose operation is the execution of a process. e.g: a python or shell script.
+ *
+ * It is able to execute an external process and is able to parse the result.
+ *
+ * It stores multiple output values and a model result vertex is required to extract a specific value.
+ */
+public class ProcessModelVertex extends DoubleVertex implements ModelVertex<DoubleTensor> {
+
+    private static final DoubleTensor MODEL_RETURN_VALUE = DoubleTensor.scalar(0.);
+
+    private String command;
```

Comment:
I wonder if you can remove the code duplication between `ProcessModelVertex` and `LambdaModelVertex`? For example both have the same fields `inputs`, `outputs`, `extractOutput`, `hasCalculated` and the same method implementations `sample`, `calculate`, `updateValues`

Reply:
in fact the two classes are so similar, do we even need both? A `ProcessModelVertex` is really just a `LambdaModelVertex` with a certain type of `executor`

Reply:
Great point - have factored out the `ProcessModelVertex` to use the `LambdaModelVertex`